NetKet/UnitaryDynamics/TimeStepper/abstract_time_stepper.hpp

```diff
@@ -31,11 +31,29 @@ template<class State>
 class AbstractTimeStepper
 {
 public:
+    /**
+     * Propagate the state x from t to t + dt.
+     * @param ode_system A callable representing the right-hand side F of the ODE.
+     *      The function is called with the current state as const reference, a
+     *      reference to the derivative dxdt which ode_system should write to and
+     *      the current time t.
+     * @param x The value of x at t.
+     * @param t The current time.
+     * @param dt The time step. Time steppers may use a smaller time step internally
+     *      (e.g., as part of an adaptive algorithm).
+     */
     virtual void Propagate(OdeSystemFunction<State> ode_system,
                            State &x,
```

Comment:
Here `State` here be a general object, right? I am already thinking at the case where we will need this to do the t-VMC and it's the variational parameters that we need to evolve, not the whole wave-function

Reply:
In the current implementation, `State` can be an arbitrary `Eigen::Matrix`-like type (which includes the `Vector` classes). If the variational parameters are stored in a different way (like a set of matrices with different shapes) we will need to either put them into a large vector or generalize the ODE implementation.